ksql-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java

```diff
@@ -255,6 +263,19 @@ public KsqlStructuredDataOutputNode cloneWithDoCreateInto(final boolean newDoCre
     );
   }
 
+  private SourceTopicProperties getSourceTopicProperties(
+      final String kafkaTopicName,
+      final KafkaTopicClient kafkaTopicClient
+  ) {
+    final Map<String, TopicDescription> topicDescriptionMap = kafkaTopicClient
+        .describeTopics(Collections.singletonList(kafkaTopicName));
+    final TopicDescription topicDescription = topicDescriptionMap.get(kafkaTopicName);
+    Objects.requireNonNull(topicDescription, "Could not fetch source topic description.");
+    final int partitionCount = topicDescription.partitions().size();
+    final short replicas = (short) topicDescription.partitions().get(0).replicas().size();
+    return new SourceTopicProperties(partitionCount, replicas);
```

Comment:
We should lazily get source topic partition and replica count, as we may not need it i.e. `SourceTopicProperties` should lazily get this info, _once_, when first requested.

Reply:
The sink topic properties are used when we call `createSinkTopic` a few lines below.

Reply:
Only _if_ they are not already supplied though, right?